owncloudApp/src/main/java/com/owncloud/android/presentation/transfers/TransferListComposeFragment.kt

```diff
@@ -0,0 +1,532 @@
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.layout.layoutId
+import androidx.compose.ui.platform.ComposeView
+import androidx.compose.ui.platform.ViewCompositionStrategy
+import androidx.compose.ui.res.colorResource
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.FontFamily
+import androidx.compose.ui.text.font.FontStyle
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextOverflow
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.fragment.app.Fragment
+import com.owncloud.android.R
+import com.owncloud.android.domain.files.model.OCFile
+import com.owncloud.android.domain.spaces.model.OCSpace
+import com.owncloud.android.domain.transfers.model.OCTransfer
+import com.owncloud.android.domain.transfers.model.TransferStatus
+import com.owncloud.android.utils.DisplayUtils
+import org.koin.androidx.viewmodel.ext.android.viewModel
+import kotlinx.coroutines.flow.collectLatest
+import java.io.File
+import android.text.format.DateUtils
+import androidx.compose.foundation.clickable
+import androidx.compose.runtime.collectAsState
+import androidx.documentfile.provider.DocumentFile
+import androidx.lifecycle.asFlow
+import androidx.work.WorkInfo
+import com.google.android.material.snackbar.Snackbar
+import com.owncloud.android.domain.transfers.model.TransferResult
+import com.owncloud.android.extensions.showMessageInSnackbar
+import com.owncloud.android.extensions.statusToStringRes
+import com.owncloud.android.lib.common.OwnCloudAccount
+import com.owncloud.android.presentation.authentication.AccountUtils
+import com.owncloud.android.ui.activity.FileActivity
+import com.owncloud.android.utils.MimetypeIconUtil
+import com.owncloud.android.workers.DownloadFileWorker
+import timber.log.Timber
+
+class TransferListComposeFragment : Fragment() {
+
+    private val transfersViewModel by viewModel<TransfersViewModel>()
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        return ComposeView(requireContext()).apply {
+            setContent {
+                setViewCompositionStrategy(ViewCompositionStrategy.DisposeOnViewTreeLifecycleDestroyed)
+                Transfers()
+            }
+        }
+    }
+
+    @Composable
+    fun Transfers() {
+        var transfersState by remember { mutableStateOf(emptyList<Pair<OCTransfer, OCSpace?>>()) }
+        val workInfos by transfersViewModel.workInfosListLiveData.asFlow().collectAsState(initial = emptyList())
```

Comment:
Don't we need this to be a `remember` block as well? And include the collecting function inside `LaunchedEffect`?

Reply:
The behavior is different in this case. The progress bar isn't updated if I put a remember variable (like `transfersState`) and the collecting function inside `LaunchedEffect`